packages/eas-cli/src/credentials/ios/appstore/authenticate.ts

```diff
@@ -40,7 +43,46 @@ export type AuthCtx = {
   fastlaneSession: string;
 };
 
+async function authenticateWithExperimentalAsync(options: Options = {}): Promise<AuthCtx> {
+  const { appleId, appleIdPassword } = await requestAppleCredentialsAsync(options);
+  log(`Authenticating to Apple Developer Portal...`); // use log instead of spinner in case we need to prompt user for 2fa
+
+  try {
+    // TODO: The password isn't required for apple-utils. Remove the local prompt when we remove traveling Fastlane.
+    await Auth.loginAsync({ username: appleId, password: appleIdPassword });
+
+    log(chalk.green('Authenticated with Apple Developer Portal successfully!'));
+
+    // Get all of the teams
+    const teams = await Teams.getTeamsAsync();
+    const team = await chooseTeamAsync(teams, options.teamId);
+
+    // Set the selected team ID internally
+    Teams.setSelectedTeamId(team.id);
```

Comment:
I totally agree with Wojtek. When I was reviewing this I almost left a comment like this but I wasn't sure if anyone thinks the same.

Reply:
We only have to manually set the team outside of the package to accommodate how Fastlane works, when we remove Fastlane this will go and team resolution will be part of the internal authentication unless specified otherwise. Similarly we resolve the password using keychain outside of apple-utils but when we remove Fastlane we'll also remove this extra step as the auth can often be resolved without retrieving the password from the keychain.

Further, the team id is passed to nearly every apple network request, Fastlane works similarly by storing the currently selected team id, but we pass the team id around because we instantiate a new Fastlane instance for seemingly every command.

Reply:
I don't see why we need to accommodate how Fastlane works. Using `traveling-fastlane`, we don't depend on this behavior and instead, we're passing the team id to every "action".

Reply:
Currently with Fastlane we do the following: 
- auth context -> traveling fastlane script -> fastlane spaceship (provides to all network requests).
In apple-utils we similarly rely on the package to provide the team id to all network requests, otherwise (almost) every network request would need to accept the Team ID like `BundleId.getAsync(query)` would become `BundleId.getAsync(teamId, query)`. I say almost because the Apple API has lots of inconsistencies which our package abstracts over.

Reply:
We could avoid passing in the team id to all requests if we had a way to create a "client" object tied to a particular team. This is a really common pattern, e.g. see `axios.create` (https://github.com/axios/axios#axioscreateconfig)  where you can specify the base URL for all requests.

This way we wouldn't depend on a global state which I consider to be a pretty serious antipattern.

Reply:
We already define a `baseUrl` for each Apple API in apple-utils, originally we set the team id in the Axios object but sometimes the API requires the team id in the [parameters](https://github.com/expo/third-party/blob/627a902c24769c74eb1c3a24dc9b18b206e0c382/packages/app-store/src/portal/PortalAPI.ts#L105) and other times it requires it in the [request body](https://github.com/expo/third-party/blob/627a902c24769c74eb1c3a24dc9b18b206e0c382/packages/app-store/src/portal/PortalAPI.ts#L78) 🙃 so we had to change to a global state to accommodate the inconsistencies.

Reply:
The reference to `axios.create` and setting a base URL was just an example.

I think that the way of using the team id is just an implementation detail of a particular request. It doesn't automatically mean that we have to have a global state for apple id. Please correct me if I'm missing sth.

Reply:
@EvanBacon 
IMO it should work sth like this:
```
import AppleUtils from '@expo/apple-utils';

const clientA = AppleUtils.createClient({ ..., teamID: 'ABC123YZ', ... });
await client.Certificate.getAsync(...); // fetches cert for team ABC123YZ

const clientB = AppleUtils.createClient({ ..., teamID: 'XYZ123AB', ... });
await clientB.Certificate.getAsync(...); // fetches cert for team XYZ123AB
```
This way we don't need the global state.